server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/RareClusterStateIT.java

```diff
@@ -371,6 +408,12 @@ public void testDelayedMappingPropagationOnReplica() throws Exception {
         // We wait on purpose to make sure that the document is not indexed because the shard operation is stalled
         // and not just because it takes time to replicate the indexing request to the replica
         Thread.sleep(100);
+        final var results = "["
+            + (putMappingResponse.isDone() ? Strings.toString(putMappingResponse.get()) : "null")
+            + ","
+            + (docIndexResponse.isDone() ? Strings.toString(docIndexResponse.get()) : "null")
+            + "]";
+        assertEquals("[null,null]", results);
         assertFalse(putMappingResponse.isDone());
         assertFalse(docIndexResponse.isDone());
```

Comment:
It looks like there is no way with hamcrest to combine multiple assertions on different objects into one.
Looks like another reason to use assertj: https://stackoverflow.com/questions/47397525/multiply-conditions-set-in-assertj-assertions